bool uncouple_req
---
string uncouple_res
